daprdocs/content/en/developing-applications/building-blocks/service-invocation/howto-invoke-services-grpc.md

```diff
@@ -0,0 +1,238 @@
+---
+type: docs
+title: "How-To: Invoke services using gRPC"
+linkTitle: "How-To: Invoke services using gRPC"
+description: "Call between services using service invocation"
+weight: 3000
+---
+
+This article describe how to deploy services each with an unique application ID, so that other services can discover and call endpoints on them using service invocation API.
+
+## Step 1: Run a gRPC server
+
+The following example is taken from the [hello world grpc-go example](https://github.com/grpc/grpc-go/tree/master/examples/helloworld).
+
+```go
+package main
+
+import (
+	"context"
+	"log"
+	"net"
+
+	"google.golang.org/grpc"
+	pb "google.golang.org/grpc/examples/helloworld/helloworld"
+)
+
+const (
+	port = ":50051"
+)
+
+// server is used to implement helloworld.GreeterServer.
+type server struct {
+	pb.UnimplementedGreeterServer
+}
+
+// SayHello implements helloworld.GreeterServer
+func (s *server) SayHello(ctx context.Context, in *pb.HelloRequest) (*pb.HelloReply, error) {
+	log.Printf("Received: %v", in.GetName())
+	return &pb.HelloReply{Message: "Hello " + in.GetName()}, nil
+}
+
+func main() {
+	lis, err := net.Listen("tcp", port)
+	if err != nil {
+		log.Fatalf("failed to listen: %v", err)
+	}
+	s := grpc.NewServer()
+	pb.RegisterGreeterServer(s, &server{})
+	log.Printf("server listening at %v", lis.Addr())
+	if err := s.Serve(lis); err != nil {
+		log.Fatalf("failed to serve: %v", err)
+	}
+}
+```
+
+This Go app implements the Greeter proto service and exposes a `SayHello` method.
+
+### Run the gRPC server using the Dapr CLI
+
+Since gRPC proxying is an opt-in feature, first enable it using a configuration file:
+
+```yaml
+apiVersion: dapr.io/v1alpha1
+kind: Configuration
+metadata:
+  name: serverconfig
+spec:
+  tracing:
+    samplingRate: "1"
+    zipkin:
+      endpointAddress: http://localhost:9411/api/v2/spans
+  features:
+    - name: proxy.grpc
+      enabled: true
+```
+
+Run the sidecar and the Go server:
+
+```bash
+dapr run --app-id server --app-protocol grpc --app-port 50051 --config config.yaml -- go run main.go
+```
+
+Using the Dapr CLI, we're assigning a unique id to the app, `server`, using the `--app-id` flag.
+
+## Step 2: Invoke the service
+
+The following example shows you how to discover the Greeter service using Dapr from a gRPC client.
+Notice that instead of invoking the target service directly at port `50051`, the client is invoking its local Dapr sidecar over port `50007` which then takes care of service discovery, telemetry, mTLS and other concerns.
```

Comment:
"which then takes care of service discovery, telemetry, mTLS and other concerns." better to say
"which then provides all the capabilities of service invocation including service discovery, tracing, mTLS and retries.

Reply:
Changed.